File: 3174-clear_digits/go/main.go
package main

import (
	"fmt"
)

func clearDigits(s string) string {
	var result string
	var toRemove int
	for i := len(s) - 1; i >= 0; i-- {
		if s[i] >= '0' && s[i] <= '9' {
			toRemove++
		} else if toRemove == 0 {
			result = string(s[i]) + result
		} else {
			toRemove--
		}
	}
	return result
}

func main() {
	s := "abc"
	result := clearDigits(s)
	fmt.Printf("Result: %#v\n", result)
	if result == "abc" {
		fmt.Println("Passed")
	} else {
		fmt.Println("Failed")
	}
}

File: 3174-clear_digits/ruby/main.rb
# @param {String} s
# @return {String}
def clear_digits(s)
  result = ""
  to_remove = 0
  (s.length - 1).downto(0) do |i|
    if s[i] >= '0' && s[i] <= '9'
      to_remove += 1
    elsif to_remove == 0
      result = s[i] + result
    else
      to_remove -= 1
    end
  end
  result
end

if __FILE__ == $0
  s = "abc"
  result = clear_digits(s)
  puts "Result: #{result}"

  if result == "abc"
    puts "Passed"
  else
    puts "Failed"
  end
end

